File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
class ResistorColorDuo {
    // Define the colors in order of their numeric value (0-9)
    private static final String[] COLORS = {
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    };

    int value(String[] colors) {
        // Get the index (value) of the first two colors
        int firstDigit = getColorCode(colors[0]);
        int secondDigit = getColorCode(colors[1]);
        
        // Combine them into a two-digit number: first * 10 + second
        return firstDigit * 10 + secondDigit;
    }

    // Helper method to find the numeric code for a color
    private int getColorCode(String color) {
        for (int i = 0; i < COLORS.length; i++) {
            if (COLORS[i].equals(color)) {
                return i;
            }
        }
        throw new IllegalArgumentException("Invalid color: " + color);
    }
}

File: solutions/ruby/resistor-color-duo/1/resistor_color_duo.rb
class ResistorColorDuo
  COLOR_CODES = {
    "black"  => 0,
    "brown"  => 1,
    "red"    => 2,
    "orange" => 3,
    "yellow" => 4,
    "green"  => 5,
    "blue"   => 6,
    "violet" => 7,
    "grey"   => 8,
    "white"  => 9
  }.freeze

  def self.value(colors)
    # Take only the first two colors
    first = COLOR_CODES[colors[0]]
    second = COLOR_CODES[colors[1]]

    # Combine into a two-digit number: 10 * first + second
    first * 10 + second
  end
end